projects/igniteui-angular/src/lib/action-strip/action-strip.component.ts

```diff
@@ -305,7 +305,7 @@ export class IgxActionStripComponent extends DisplayDensityBase implements After
     public hide(): void {
         this.hidden = true;
         this.closeMenu();
-        if (this.context && this.context.element) {
+        if (this.context && this.context.element && !!this._originalParent) {
```

Comment:
As discussed, as Angular can initialize the component before it's attached to the DOM - the ctor is a bad place to gather the original parent. To mirror this, it should be available to save during `show` actually.
This also got me thinking that it's probably not a bad idea to keep the old detach behavior as fallback if there's no parent - I think that might be technically possible if you dynamically create the actions and don't attach them to a view, right?

Reply:
Yes when created dynamically the `originalParent` would be null on show, but when it is rendered at least in Angular it would go through the AfterViewInit hook and it will set the `originalParent `. I've added the old behavior just in case though.